packages/manager/apps/web-office/src/pages/dashboard/users/ModalEditUsers.component.tsx

```diff
@@ -0,0 +1,258 @@
+  } = useOfficeUserDetail(activationEmail, licencePrepaidName);
+
+  const [isLoading, setIsLoading] = useState(
+    !userDetail || isUserDetailLoading,
+  );
+
+  const {
+    control,
+    setValue,
+    getValues,
+    formState: { isDirty, isValid, errors },
+  } = useForm({
+    defaultValues: {
+      firstname: userDetail?.firstName || '',
+      lastname: userDetail?.lastName || '',
+      login: userDetail?.activationEmail.split('@')[0] || '',
+    },
+    mode: 'onChange',
+    resolver: zodResolver(EDIT_USERS_FORM_SCHEMA),
+  });
+
+  useEffect(() => {
+    if (userDetail && !isUserDetailLoading) {
+      setValue('firstname', userDetail.firstName);
+      setValue('lastname', userDetail.lastName);
+      setValue('login', userDetail.activationEmail.split('@')[0]);
+      setIsLoading(false);
+    }
+  }, [userDetail, isUserDetailLoading]);
+
+  const { mutate: editUser, isPending: isSending } = useMutation({
+    mutationFn: (params: UserParamsType) => {
+      return licencePrepaidName
+        ? putOfficeLicenseDetails(licencePrepaidName, params)
+        : putOfficeUserDetail(selectedServiceName, activationEmail, params);
+    },
+    onSuccess: () => {
+      addSuccess(
+        <OdsText preset={ODS_TEXT_PRESET.paragraph}>
+          {t('dashboard_users_edit_message_success')}
+        </OdsText>,
+        true,
+      );
+    },
+    onError: (error: ApiError) => {
+      addError(
+        <OdsText preset={ODS_TEXT_PRESET.paragraph}>
+          {t('dashboard_users_edit_message_error', {
+            error: error.response?.data?.message,
+          })}
+        </OdsText>,
+        true,
+      );
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries({
+        queryKey: [
+          licencePrepaidName
+            ? getOfficeLicenseQueryKey(selectedServiceName)
+            : getOfficeUsersQueryKey(selectedServiceName),
+        ],
+      });
+      onClose();
+    },
+  });
+
+  const handleSaveClick = () => {
+    const { firstname, lastname, login } = getValues();
+    editUser({
+      firstName: firstname,
+      lastName: lastname,
+      activationEmail: `${login}@${userDetail.activationEmail.split('@')[1]}`,
+    });
+  };
+
+  const handleCancelClick = () => {
+    onClose();
+  };
+
+  return (
+    <form>
+      <Modal
+        isOpen
+        title={t('dashboard_users_edit_title')}
+        color={ODS_MODAL_COLOR.information}
+        onClose={onClose}
+        isDismissible
+        isLoading={isLoading}
+        secondaryButton={{
+          label: t('common:cta_cancel'),
+          action: handleCancelClick,
+        }}
+        primaryButton={{
+          testid: 'edit-btn',
+          variant: ODS_BUTTON_VARIANT.default,
+          label: t('common:cta_confirm'),
+          isDisabled: !isDirty || !isValid,
+          isLoading: isSending,
+          action: handleSaveClick,
+        }}
+      >
+        <>
+          <OdsText preset={ODS_TEXT_PRESET.paragraph} className="mt-5">
```

Comment:
if you can apply a gap between the fields instead of putting margin top on each one it would be better imo

Reply:
It's fixed